haystack/nodes/connector/crawler.py

```diff
@@ -101,21 +101,21 @@ def __init__(
 
         IN_COLAB = "google.colab" in sys.modules
         IN_AZUREML = os.environ.get("AZUREML_ENVIRONMENT_IMAGE", None) == "True"
-        IS_ROOT = sys.platform not in ["win32", "cygwin"] and os.geteuid() == 0  # type: ignore   # This is a mypy issue of sorts, that fails on Windows.
+        IN_WINDOWS = sys.platform in ["win32", "cygwin"]
+        IS_ROOT = not IN_WINDOWS and os.geteuid() == 0  # type: ignore   # This is a mypy issue of sorts, that fails on Windows.
 
         if webdriver_options is None:
             webdriver_options = ["--headless", "--disable-gpu", "--disable-dev-shm-usage", "--single-process"]
-        elif "--headless" not in webdriver_options:
-            webdriver_options.append("--headless")
-
-        if IS_ROOT and "--no-sandbox" not in webdriver_options:
+        webdriver_options.append("--headless")
```

Comment:
I think we should first check if `--headless` is in `webdriver_options` before appending it otherwise we run the risk of specifying it twice when `webdriver_options=None` is the input.

Reply:
**Update: ** Sorry I didn't see the `set` usage below. Please disregard this comment. I should really read all the changes first XD